extra_geom/detectors.py

```diff
@@ -768,6 +769,140 @@ def offset(self, shift, *, modules=np.s_[:], tiles=np.s_[:]):
         ])
 
 
+class GenericGeometry(DetectorGeometryBase):
+    """A generic detector layout based either on the CrystFEL geom file or on a set of parameters.
+
+    The coordinates used in this class are 3D (x, y, z), and represent metres.
+
+    The :attr:`expected_data_shape` is a following triple :
+
+        1. the number of modules which is the length of the :attr:`corner_coordinates` array
+        2. the number of tiles in a module along the slow-scan direction multiplied by
+           the number of slow-scan pixels per tile :attr:`frag_ss_pixels`
+        3. the number of tiles in a module along the fast-scan direction multiplied by
+           the number of fast-scan pixels per tile :attr:`frag_fs_pixels`
+    """
+    detector_type_name = 'Generic Detector'
+
+    @classmethod
+    def from_simple_description(cls, pixel_size: float, slow_pixels: int, fast_pixels: int,
+                                corner_coordinates: List[np.ndarray] = [np.zeros(3)],
+                                ss_vec: np.ndarray = np.array([1, 0, 0]),
+                                fs_vec: np.ndarray = np.array([0, 1, 0]),
+                                n_tiles_per_module: int = 1,
+                                tile_gap: float = None,
+                                tile_vec: np.ndarray = None,
+                                ):
+        """ Creates a generic detector from a dictionary.
+
+        Parameters
+        __________
+
+        pixel_size: float
+            the size of a pixel in meters (reversed CrystFEL's `res`)
+        slow_pixels, fast_pixels: int
+            the size of a tile along the slow- and the fast-scan axes
+        corner_coordinates: ndarray
+            3D coordinates of the first pixel of each module
+        ss_vec, fs_vec: ndarray
+            3D vectors of the slow- and the fast-scan directions in the lab coordinates (the X-axis
```

Comment:
I still think this should be clearer that these are in units of pixel size, i.e. magnitude 1. 'Lab coordinates' implies they're in regular units like metres (at least to my mind).

Reply:
I tried to fix this, please see. I also removed unused `geom`-files from test directory.